File: ruby/task2/4mod2.rb
def count_digits(nums)
    count = 0
    nums.each do |num|
      num = num.abs 
      digits = 0 
      if num == 0
        digits = 1 
      else
        while num != 0
          num /= 10
          digits += 1
        end
      end
      count += 1 if digits.odd? #Встроенная проверка на четность (odd)
    end
  
    return count
  end
  
  # Основная программа
  puts "Введите размер массива:"
  size = gets.chomp.to_i  # Переводим число в int
  numbers = []
  (1..size).each do |i| # Ввод массива
    print "#{i} Элемент массива: "
    num = gets.chomp.to_i
    numbers << num # Добавление в массив 
  end
  
  # Вывод
  result = count_digits(numbers)
  puts "#{result} Чисел(а) содержат нечетное количество цифр"
  

File: golang/task2/4mod2.go
package main

import (
	"fmt"
)

func countDigits(slice []int) int { // slice - то же самое что & в C++
	count := 0
	for i := 0; i < len(slice); i++ {
		n := slice[i]
		digits := 0
		if n < 0 {
			n = -n
		} //если отрицательное число
		if n == 0 {
			digits = 1
		}
		if n > 0 {
			for n != 0 { //аналогия с while
				n /= 10
				digits++
			}
		}
		if digits%2 != 0 {
			count++
		}
	}
	return count
}

func main() {
	var n int
	fmt.Println("Введите количество элементов")
	fmt.Scan(&n)

	slice := make([]int, n)
	fmt.Println("Введите", n, "количество целых чисел")

	for i := 0; i < n; i++ {
		fmt.Scan(&slice[i])
	}

	result := countDigits(slice)
	fmt.Println("Количество числе с нечетными цифрами", result)
}
